tests/test_injection.py

```diff
@@ -37,7 +39,13 @@ def __init__(self, crawler):
             self.crawler = crawler
 
         def __call__(self, to_provide):
-            return [cls(content) if content else cls() for cls in classes]
+            result = []
+            for cls in to_provide:
+                obj = cls(content) if content else cls()
+                if metadata := getattr(cls, "__metadata__", None):
+                    obj = AnnotatedResult(obj, metadata)
```

Comment:
We need the same change in all real providers if we want to support annotating their provided classes. Also, most of those providers currently don't check `to_provide`, assuming it's the same as the declared provided class(es) so they will need logic improvement. This needs to include support for returning several instances when they are requested.